moped-api/tests/test_users.py

```diff
@@ -438,14 +475,70 @@ def test_deletes_user(
         user_to_delete = mock_users[0]["email"]
 
         # Delete user request
-        response = self.client.delete(f"/users/{user_to_delete}",)
+        response = self.client.delete(
+            f"/users/{user_to_delete}",
+        )
         response_dict = self.parse_response(response.data)
 
         # Get user list to compare length
         user_list_response = self.client.get("/users/")
         user_list = self.parse_response(user_list_response.data)
 
         assert isinstance(response_dict, dict)
-        assert response_dict["ResponseMetadata"]["HTTPStatusCode"] == 200
+        assert isinstance(response_dict.get("success", None), dict)
         assert len(user_list) == len(mock_users) - 1
 
+    @patch("flask_cognito._cognito_auth_required")
+    @patch("users.users.is_valid_user")
+    @patch("users.users.is_users_password")
+    @patch("users.users.is_valid_user_password")
+    def test_update_user_password(
+        self,
+        mock_is_valid_user_password,
+        mock_is_users_password,
+        mock_is_valid_user,
+        mock_cognito_auth_required,
+        create_user_pool,
+    ):
+        """Test update password route."""
+        # Mock valid user check and claims loaded from DynamoDB
+        mock_is_valid_user.return_value = True
+        mock_is_users_password.return_value = True
+        mock_is_valid_user_password.return_value = True, None
+
+        # Patch normalize claims
+        claims = create_user_claims()
+        claims["https://hasura.io/jwt/claims"] = json.dumps(
+            claims["https://hasura.io/jwt/claims"]
+        )
+        _get_current_object = Mock(return_value=claims)
+        patch(
+            "claims.current_cognito_jwt",
+            Mock(_get_current_object=_get_current_object),
+        ).start()
+
+        # Mock user pool
+        user_pool_dict = create_user_pool
+        user_pool_id = user_pool_dict["user_pool_id"]
+
+        # Patch the user pool id with the mock pool id
+        patch("users.users.USER_POOL", new=user_pool_id).start()
+
+        # Prepare the payload for the request
+        user_to_update_password = mock_users[0]["email"]
+        new_password_json_payload = json.dumps({"password": "nUPaSSword"})
+
+        # Delete user request
+        response = self.client.put(
+            f"/users/{user_to_update_password}/password",
+            data=new_password_json_payload,
+            content_type="application/json",
+        )
+        response_dict = self.parse_response(response.data)
+
+        # Get user list to compare length
+        user_list_response = self.client.get("/users/")
+        user_list = self.parse_response(user_list_response.data)
+
+        assert isinstance(response_dict, dict)
+        assert isinstance(response_dict.get("success", None), dict)
```

Comment:
Thanks for your cleverness Mike! :) I enjoy learning from you

Reply:
Thanks! There is a lot of patching in this branch to get through the routes so if we ever decide we need more coverage for the DynamoDB actions then hopefully it will provide a good base for that. 👍